assignment/tool/src/webapp/js/assignments.js

```diff
@@ -969,6 +969,26 @@ ASN.changeVisibleDate = function()
 	}
 }
 
+ASN.changeResubmissionDate = function()
+{
+  if(document.getElementById("allowResToggle").checked)
+  {
+    document.getElementById('allowResubmitNumber').style.display = 'block';
+    document.getElementById('allowResubmitTime').style.display = 'block';
+    document.getElementById('resubmitNotification').style.display = 'block';
+    const closeDatePicker = document.getElementById('closedate');
+    const resubmitDatePicker = document.getElementById('resubmissiondate');
+    resubmitDatePicker.isoDate = closeDatePicker.isoDate;
+    ASN.resizeFrame();
+  }
+  else
+  {
+    document.getElementById('allowResubmitNumber').style.display = 'none';
+    document.getElementById('allowResubmitTime').style.display = 'none';
+    document.getElementById('resubmitNotification').style.display = 'none';
+  }
+}
+
```

Comment:
```suggestion
ASN.changeResubmissionDate = function() {

  const displayStyle = document.getElementById("allowResToggle").checked ? 'block' : 'none';

  const elementIds = ['allowResubmitNumber', 'allowResubmitTime', 'resubmitNotification'];

  elementIds.forEach(id => document.getElementById(id).style.display = displayStyle);

  if (displayStyle === 'block') {
  
    document.getElementById('resubmissiondate').isoDate = document.getElementById('closedate').isoDate;
  }
};

```

Reply:
i smell chatgpt ;)

Reply:
ultimately, the way this is done with three separate DIVs being hidden/shown is bad. I considered consolidating it down to one container DIV but wasn't ready to spend that much time on this one.